cbits/dns.c

```diff
@@ -0,0 +1,84 @@
+#include <winsock2.h>
+#include <iphlpapi.h>
+#include <string.h>
+#include <windows.h>
+#include <string.h>
+#include "dns.h"
+
+#define MALLOC(x) HeapAlloc(GetProcessHeap(), 0, (x))
+#define FREE(x) HeapFree(GetProcessHeap(), 0, (x))
+
+dns_t* getWindowsDefDnsServers(void) {
```

Comment:
The typical pattern for C functions like this that return both an error code and another result (the ',' sep string here) is to have the function return the error code and for the caller to pass the output buffer and size as function arguments. Then the caller manages the output buffer and the C function does not have to malloc or free the buffer (or decide how big it should be). This would avoid the current problems you have below where the buffers and output structure are allocated but never freed. There would be no allocation of results at all (just the `FIXED_INFO` that is locally allocated and decallocated).